https://bugs.gentoo.org/499086
https://rt.openssl.org/Ticket/Display.html?id=3333&user=guest&pass=guest

when gcc is given a .s file and told to preprocess it, it outputs nothing

From a2976461784ce463fc7f336cd0dce607d21c2fad Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sat, 25 Jan 2014 05:44:47 -0500
Subject: [PATCH] Revert "Make Makefiles OSF-make-friendly."

This reverts commit d1cf23ac86c05b22b8780e2c03b67230564d2d34.
---
 crypto/Makefile       | 4 +---
 crypto/bn/Makefile    | 4 +---
 crypto/evp/Makefile   | 2 +-
 crypto/modes/Makefile | 5 +----
 crypto/sha/Makefile   | 4 +---
 util/shlib_wrap.sh    | 6 +-----
 6 files changed, 6 insertions(+), 19 deletions(-)

diff --git a/crypto/Makefile b/crypto/Makefile
index b253f50..1de9d5f 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -86,9 +86,7 @@ ia64cpuid.s:	ia64cpuid.S;	$(CC) $(CFLAGS) -E ia64cpuid.S > $@
 ppccpuid.s:	ppccpuid.pl;	$(PERL) ppccpuid.pl $(PERLASM_SCHEME) $@
 pariscid.s:	pariscid.pl;	$(PERL) pariscid.pl $(PERLASM_SCHEME) $@
 alphacpuid.s:	alphacpuid.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
-	$(PERL) alphacpuid.pl > $$preproc && \
-	$(CC) -E $$preproc > $@ && rm $$preproc)
+	$(PERL) $< | $(CC) -E - | tee $@ > /dev/null
 
 subdirs:
 	@target=all; $(RECURSIVE_MAKE)
diff --git a/crypto/bn/Makefile b/crypto/bn/Makefile
index b62b676..6c03363 100644
--- a/crypto/bn/Makefile
+++ b/crypto/bn/Makefile
@@ -136,9 +136,7 @@ ppc-mont.s:	asm/ppc-mont.pl;$(PERL) asm/ppc-mont.pl $(PERLASM_SCHEME) $@
 ppc64-mont.s:	asm/ppc64-mont.pl;$(PERL) asm/ppc64-mont.pl $(PERLASM_SCHEME) $@
 
 alpha-mont.s:	asm/alpha-mont.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
-	$(PERL) asm/alpha-mont.pl > $$preproc && \
-	$(CC) -E $$preproc > $@ && rm $$preproc)
+	$(PERL) $< | $(CC) -E - | tee $@ > /dev/null
 
 # GNU make "catch all"
 %-mont.S:	asm/%-mont.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
diff --git a/crypto/modes/Makefile b/crypto/modes/Makefile
index ce0dcd6..88ac65e 100644
--- a/crypto/modes/Makefile
+++ b/crypto/modes/Makefile
@@ -55,10 +55,7 @@ aesni-gcm-x86_64.s:	asm/aesni-gcm-x86_64.pl
 ghash-sparcv9.s:	asm/ghash-sparcv9.pl
 	$(PERL) asm/ghash-sparcv9.pl $@ $(CFLAGS)
 ghash-alpha.s:	asm/ghash-alpha.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
-	$(PERL) asm/ghash-alpha.pl > $$preproc && \
-	$(CC) -E $$preproc > $@ && rm $$preproc)
-
+	$(PERL) $< | $(CC) -E - | tee $@ > /dev/null
 ghash-parisc.s:	asm/ghash-parisc.pl
 	$(PERL) asm/ghash-parisc.pl $(PERLASM_SCHEME) $@
 
diff --git a/crypto/sha/Makefile b/crypto/sha/Makefile
index 64eab6c..63fba69 100644
--- a/crypto/sha/Makefile
+++ b/crypto/sha/Makefile
@@ -60,9 +60,7 @@ sha256-armv4.S: asm/sha256-armv4.pl
 	$(PERL) $< $(PERLASM_SCHEME) $@
 
 sha1-alpha.s:	asm/sha1-alpha.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
-	$(PERL) asm/sha1-alpha.pl > $$preproc && \
-	$(CC) -E $$preproc > $@ && rm $$preproc)
+	$(PERL) $< | $(CC) -E - | tee $@ > /dev/null
 
 # Solaris make has to be explicitly told
 sha1-x86_64.s:	asm/sha1-x86_64.pl;	$(PERL) asm/sha1-x86_64.pl $(PERLASM_SCHEME) > $@
-- 
1.8.5.3

